From 6b2382da4e7345be6ce086d7acb805e91d8333d4 Mon Sep 17 00:00:00 2001
From: Thomas Stewart <stewart@schrodinger.com>
Date: Mon, 3 May 2021 13:21:29 -0400
Subject: [PATCH] PYMOL-3664 Using Bio.Align instead of Bio.SubsMat -
 Bio.SubsMat.MatrixInfo is depricated - Now trying to import
 substitution_matrices   from Bio.Align first

---
 modules/pymol/seqalign.py | 7 ++++++-
 1 file changed, 6 insertions(+), 1 deletion(-)

diff --git a/modules/pymol/seqalign.py b/modules/pymol/seqalign.py
index 99b65c4b..2c47b3d8 100644
--- a/modules/pymol/seqalign.py
+++ b/modules/pymol/seqalign.py
@@ -21,7 +21,12 @@ DESCRIPTION
     '''
     from Bio import pairwise2
     from Bio.Align import MultipleSeqAlignment
-    from Bio.SubsMat.MatrixInfo import blosum62
+    try:
+        from Bio.Align import substitution_matrices
+    except ImportError:
+        from Bio.SubsMat.MatrixInfo import blosum62
+    else:
+        blosum62 = substitution_matrices.load("BLOSUM62")
 
     def match_callback(c1, c2):
         return blosum62.get((c1, c2), 1 if c1 == c2 else -4)
-- 
2.32.0

